components/event-publisher-proxy/pkg/handler/http/handler_test.go

```diff
@@ -460,14 +464,27 @@ func setupTestResources(t *testing.T, port, maxRequestSize int, applicationName,
 
 func validateApplicationName(appName string) testingutils.Validator {
 	return func(r *http.Request) error {
+		if r.Header.Get(cev2http.ContentType) == cloudevents.CEStructuredMode {
+			// CE structured mode
+			message := cev2http.NewMessageFromHttpRequest(r)
+			defer func() { _ = message.Finish(nil) }()
+			if event, err := binding.ToEvent(context.Background(), message); err != nil {
+				return err
+			} else if strings.Contains(event.Type(), appName) {
+				return nil
+			} else {
+				return errors.New(fmt.Sprintf("expected the event-type [%s] to contain the application name [%s] for structured mode", event.Type(), appName))
```

Comment:
`errors.Newf("%s", str)` where the import is `"gopkg.in/errgo.v2/fmt/errors"` can be handy.

Reply:
Avoid to change the go.mod for now.